purrsoft/src/scenes/Voluntari/component.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  TextField,
+  IconButton,
+  Container,
+  Grid,
+  InputAdornment,
+  Typography,
+} from '@mui/material';
+import { Search, Mic } from '@mui/icons-material';
+import { CustomCard } from '../../components/CustomCard';
+import { useTheme } from '@mui/material/styles';
+import { useGetVolunteersQuery } from '../../store';
+
+export const Voluntari = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const { data, error, isLoading } = useGetVolunteersQuery();
+
+  const theme = useTheme();
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value.toLowerCase());
+  };
+
+  const filteredVolunteers = 
+    data?.records.filter(
+        (volunteer) => 
+            (volunteer.firstName.toLowerCase() + ' ' +
+            volunteer.lastName.toLowerCase()).includes(searchTerm)
+    ) || [];
+
+  if (isLoading) {
```

Comment:
Maybe instead of only returning a div with the text loading, use a CircularProgress or LinearProgress (see that everyone has the same Progress component choice for it, I've left the same comment to Florin). 
I would say it is more user friendly.